File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<int>>group;
        for(int i=0;i<strs.size();i++)
        {
            string temp=strs[i];
            sort(temp.begin(),temp.end());
            group[temp].push_back(i);
        }
        vector<vector<string>>result;
        for(auto g:group)
        {
            vector<string>row;
            for(int i=0;i<g.second.size();i++)
            {
                row.push_back(strs[g.second[i]]);
            }
            result.push_back(row);
        }
        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> mp = new HashMap<>();
        for(int i=0;i<strs.length;i++)
        {
            char[] chr = strs[i].toCharArray();
            Arrays.sort(chr);
            String nstr = new String(chr);
            if(!mp.containsKey(nstr))
            {
                mp.put(nstr,new ArrayList<String>());
            }
            mp.get(nstr).add(strs[i]);
        }
        List<List<String>>result = new ArrayList<>();
        for(List<String> group : mp.values())
        {
            result.add(group);
        }
        return result;
    }
}